packages/learn-card-core/src/wallet/base/wallet.ts

```diff
@@ -139,6 +139,19 @@ const addCachingToStorePlane = <
     },
     // TODO: Add caching to uploadMany
     ...('uploadMany' in plane ? { uploadMany: plane.uploadMany } : {}),
+    ...('uploadEncrypted' in plane
+        ? {
+              uploadEncrypted: async (_learnCard, vc, params, { cache = 'cache-first' } = {}) => {
+                  const uri = await plane.uploadEncrypted?.(_learnCard, vc, params);
```

Comment:
Yeah, this is one of those annoying limitations of the TS compiler in its current form. You can clearly see that we are explicitly checking for `uploadEncrypted` in the plane here, but it still thinks it can't be certain because that call is within the scope of an arbitrary function, so technically, according to TS, it's possible that the plane object could be changed to remove that method between when this call happens and when the method is actually called.

It's annoying, but technically correct according to the types it is given. Another thing I've thought about doing here is to always implement the optional methods and then automatically fallback or error based on some passed in config or something, which is probably the right solution and would free us from this weird TS pedantry